runtime/js/99_main.js

```diff
@@ -657,118 +657,124 @@ function bootstrapMainRuntime(runtimeOptions) {
   performance.setTimeOrigin(DateNow());
   globalThis_ = globalThis;
 
-  // Remove bootstrapping data from the global scope
-  delete globalThis.__bootstrap;
-  delete globalThis.bootstrap;
-  delete globalThis.nodeBootstrap;
-  hasBootstrapped = true;
-
-  // If the `--location` flag isn't set, make `globalThis.location` `undefined` and
-  // writable, so that they can mock it themselves if they like. If the flag was
-  // set, define `globalThis.location`, using the provided value.
-  if (location_ == null) {
-    mainRuntimeGlobalProperties.location = {
-      writable: true,
-    };
-  } else {
-    location.setLocationHref(location_);
-  }
-
-  exposeUnstableFeaturesForWindowOrWorkerGlobalScope({
-    unstableFlag,
-    unstableFeatures,
-  });
-  ObjectDefineProperties(globalThis, mainRuntimeGlobalProperties);
-  ObjectDefineProperties(globalThis, {
-    // TODO(bartlomieju): in the future we might want to change the
-    // behavior of setting `name` to actually update the process name.
-    // Empty string matches what browsers do.
-    name: core.propWritable(""),
-    close: core.propWritable(windowClose),
-    closed: core.propGetterOnly(() => windowIsClosing),
-  });
-  ObjectSetPrototypeOf(globalThis, Window.prototype);
+  // We don't want to set up the global object if we're just warming up
+  // the bootstrap process.
+  //
+  // TODO(@littledivy): Investigate more into warming up more parts.
+  if (!warmup) {
```

Comment:
Maybe invert it and exit early?

Reply:
For some reason, inverting the control flow does not cause the code inside the `if` to compile.

Reply:
I see, maybe leave a comment about it?